flexget/plugins/input/rss.py

```diff
@@ -184,6 +185,26 @@ def process_invalid_content(self, task, data, url):
             f.write(data)
         log.critical('I have saved the invalid content to %s for you to view', filepath)
 
+    def escape_content(self, content):
+        valid_escapes = ['&quot;', '&apos;', '&lt;', '&gt;', '&amp;']
+        future_result = []
+        in_cdata_block = False
+        
+        for idx, char in enumerate(content):
+            char = chr(char)
+            if not in_cdata_block and char == '&':
+                for escape in valid_escapes:
+                    if content[idx:idx+len(escape)] == escape:
+                        break
+                else:
+                    char = '&amp;'
+            elif not in_cdata_block and char == '<' and content[idx:idx+9] == '<![CDATA[':
+                in_cdata_block = True
+            elif in_cdata_block and char == ']' and content[idx:idx+3] == ']]>':
+                should_change = False
```

Comment:
This doesn't really do anything (you probably forgot to change it).  You probably want to like this instead, since it becomes a little more complex to skip ahead 3 characters while enumerating:

```
elif in_cdata_block and char == ']' and content[idx-2:idx+1] == ']]>':
     in_cdata_block= False
```

Reply:
Thank you for comment and finding performance issue